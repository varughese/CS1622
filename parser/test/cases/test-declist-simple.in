int x;

void foo(void)
{
}

int bar(int x, int y)
{
    /* TODO return */
}

int y[10];

int baz(void)
{
    /* TODO return */
}

int z;
int w;
int j[9];
int k;

void main(void)
{
}
