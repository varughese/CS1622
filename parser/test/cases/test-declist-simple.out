[program
  [var-declaration [int] [x]]
  [fun-declaration
    [void]
    [foo]
    [params]
    [compound-stmt 
    ]
  ]
  [fun-declaration
    [int]
    [bar]
    [params 
      [param [int] [x]] 
      [param [int] [y]]]
    [compound-stmt 
    ]
  ]
  [var-declaration [int] [y] [10]]
  [fun-declaration
    [int]
    [baz]
    [params]
    [compound-stmt 
    ]
  ]
  [var-declaration [int] [z]]
  [var-declaration [int] [w]]
  [var-declaration [int] [j] [9]]
  [var-declaration [int] [k]]
  [fun-declaration
    [void]
    [main]
    [params]
    [compound-stmt 
    ]
  ]
]
